languages/go/exercises/concept/basic-strings/basic_strings_test.go

```diff
@@ -0,0 +1,201 @@
+package basic_strings
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestMessage(t *testing.T) {
+	type args struct {
+		line string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "Extract message from error message",
+			args: args{
+				line: "[ERROR]: Stack overflow",
+			},
+			want: "Stack overflow",
+		},
+		{
+			name: "Extract message from warning message",
+			args: args{
+				line: "[WARNING]: Disk almost full",
+			},
+			want: "Disk almost full",
+		},
+		{
+			name: "Extract message from info message",
+			args: args{
+				line: "[INFO]: File moved",
+			},
+			want: "File moved",
+		},
+		{
+			name: "Extract message without extra whitespace",
+			args: args{
+				line: "[WARNING]:   \tTimezone not set  \r\n",
+			},
+			want: "Timezone not set",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := Message(tt.args.line); got != tt.want {
+				t.Errorf("Message(\"%v\") = \"%v\", want \"%v\"", escapeWhiteSpace(tt.args.line), got, tt.want)
```

Comment:
Shouldn't we flip these to `%s` instead of `%v` since we know the types?

Reply:
Same comment throughout when we know what the types are...

Reply:
have been wondering that, too. I'll fix it.